crates/ruff/src/rules/flake8_bandit/helpers.rs

```diff
@@ -2,9 +2,7 @@ use rustpython_parser::ast::{Constant, Expr, ExprKind};
 
 use crate::checkers::ast::Checker;
 
-const PASSWORD_NAMES: [&str; 7] = [
-    "password", "pass", "passwd", "pwd", "secret", "token", "secrete",
-];
+const PASSWORD_NAMES: [&str; 6] = ["password", "passwd", "pwd", "secret", "token", "api_key"];
```

Comment:
I'm wondering if, instead, we should consider using a regex to match Bandit's behavior?

https://github.com/PyCQA/bandit/blob/a9eaafa53ae73c1a66172373128cb19bb7b82a32/bandit/plugins/general_hardcoded_password.py#L13-L16

Reply:
Was going to say the same thing. This is a small improvement but doesn't solve the main problem I raised in the issue - that the current substring match causes matching parts of larger words.

Reply:
@charliermarsh would regex be in the scope of the linked issue? if so, i'll implement it in this pr.

Reply:
Yeah I believe it's within scope.